rclpy/src/rclpy/_rclpy.c

```diff
@@ -60,6 +67,15 @@ rclpy_create_node(PyObject * Py_UNUSED(self), PyObject * args)
   return pynode;
 }
 
+/// Create a publisher
+/*
+ * \param[in] pynode Capsule pointing to the node to add the publisher to
+ * \param[in] pymsg_type ROS 2 type of the message this publisher will send
```

Comment:
Yeah I didn't know how to phrase it, that's a tricky one in the sense that the user passes a message module to the function (e.g. String) but we expect this module to already have the typesupport imported in it.
But they don't pass the typesupport structure itself given that this structure is extracted from the class in https://github.com/ros2/rclpy/blob/f36710dec992e8e1d0edb38477c0edf2e5086392/rclpy/src/rclpy/_rclpy.c#L103 
I'll try to rephrase this to make it clearer